packages/sanity/src/core/config/configPropertyReducers.ts

```diff
@@ -395,6 +395,52 @@ export const eventsAPIReducer = (opts: {
   return result
 }
 
+export const mediaLibraryEnabledReducer = (opts: {
+  config: PluginOptions
+  initialValue: boolean
+}): boolean => {
+  const {config, initialValue} = opts
+  const flattenedConfig = flattenConfig(config, [])
+
+  const result = flattenedConfig.reduce((acc, {config: innerConfig}) => {
+    const resolver = innerConfig.mediaLibrary?.enabled
+
+    if (!resolver && typeof resolver !== 'boolean') return acc
+    if (typeof resolver === 'boolean') return resolver
+
+    throw new Error(
+      `Expected \`mediaLibrary.enabled\` to be a boolean, but received ${getPrintableType(
+        resolver,
+      )}`,
+    )
+  }, initialValue)
+
+  return result
+}
+
+export const mediaLibraryLibraryIdReducer = (opts: {
+  config: PluginOptions
+  initialValue: string | undefined
+}): string | undefined => {
+  const {config, initialValue} = opts
+  const flattenedConfig = flattenConfig(config, [])
+
+  const result = flattenedConfig.reduce((acc, {config: innerConfig}) => {
+    const resolver = innerConfig.mediaLibrary?.libraryId
+
+    if (!resolver && typeof resolver !== 'string') return acc
+    if (typeof resolver === 'string') return resolver
+
+    throw new Error(
+      `Expected \`mediaLibrary.libraryId\` to be a string, but received ${getPrintableType(
+        resolver,
+      )}`,
+    )
+  }, initialValue)
+
+  return result
```

Comment:
Do we need to reduce here? 
```suggestion
  for (const {config: innerConfig} of flattenedConfig) {
  	const libraryId = innerConfig.mediaLibrary?.libraryId
  	if (!libraryId && typeof libraryId !== 'string') {
  		continue
  	}
  	if (typeof libraryId === 'string') {
  		return libraryId
  	}
  	
  	 throw new Error(
      `Expected \`mediaLibrary.libraryId\` to be a string, but received ${getPrintableType(
        resolver,
      )}`,
    )
  }

  return initialValue
```

Reply:
If we are to support these config options throughout the plugin chain, I think we need them.

Reply:
I first thought you meant the config recucing function in general. 

As for the actual `.reduce` I just copied what we do elsewhere. I think it could be getting a function as well, you see how it actually returns something if it's not a string.

Reply:
Will have another look at this.

Reply:
I agree with PK's initial approach here (using a reducer), purely from the standpoint of maintaining consistency with how Studio resolved other configs.

Reply:
Yes, there's a lot of these around, so I think it's better to just follow the same pattern as we usually do in this code.